server/sonar-server/src/test/java/org/sonar/server/qualitygate/ws/DestroyActionTest.java

```diff
@@ -111,17 +128,30 @@ public void should_delete_quality_gate_if_non_default_when_a_default_exist() {
     Long toDeleteQualityGateId = qualityGate.getId();
     assertThat(db.getDbClient().qualityGateDao().selectById(dbSession, toDeleteQualityGateId)).isNotNull();
 
-    QualityGateDto defaultqualityGate = db.qualityGates().insertQualityGate(qg -> qg.setName("Default"));
-    Long defaultQualityGateId = defaultqualityGate.getId();
-    db.properties().insertProperty(new PropertyDto().setKey("sonar.qualitygate").setValue(Long.toString(defaultQualityGateId)));
-    assertThat(db.getDbClient().qualityGateDao().selectById(dbSession, defaultQualityGateId)).isNotNull();
+    QualityGateDto defaultQualityGate = db.qualityGates().insertQualityGate(qg -> qg.setName("Default"));
+    db.qualityGates().setDefaultQualityGate(defaultQualityGate);
 
     ws.newRequest()
       .setParam(PARAM_ID, valueOf(toDeleteQualityGateId))
       .execute();
 
     assertThat(db.getDbClient().qualityGateDao().selectById(dbSession, toDeleteQualityGateId)).isNull();
-    assertThat(db.getDbClient().qualityGateDao().selectById(dbSession, defaultQualityGateId)).isNotNull();
+  }
+
+  @Test
+  public void should_not_delete_built_in_quality_gate() {
+    userSession.addPermission(ADMINISTER_QUALITY_GATES, db.getDefaultOrganization());
+    QualityGateDto qualityGate = db.qualityGates().insertQualityGate(qg -> qg.setBuiltIn(true));
+    Long qualityGateId = qualityGate.getId();
+
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage(format("Operation forbidden for built-in Quality Gate '%s'", qualityGate.getName()));
+
+    ws.newRequest()
+      .setParam(PARAM_ID, valueOf(qualityGateId))
+      .execute();
+
+    assertThat(db.getDbClient().qualityGateDao().selectById(dbSession, qualityGateId)).isNotNull();
   }
 
```

Comment:
Please add the following missing tests : 
- Fail on unknown quality gate
- Fail when missing qgate administer permission

Reply:
@gjambet you have forgot this comment

Reply:
I've added the missing UT